flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1441,6 +1468,59 @@ private String buildShowCreateTableRow(
         return sb.toString();
     }
 
+    /** Show create view statement only for views. */
+    private String buildShowCreateViewRow(
+            ResolvedCatalogBaseTable<?> view,
+            ObjectIdentifier viewIdentifier,
+            boolean isTemporary) {
+        final String printIndent = "  ";
+        CatalogBaseTable.TableKind kind = view.getTableKind();
+        if (kind != CatalogBaseTable.TableKind.VIEW) {
+            throw new TableException(
+                    String.format(
+                            "SHOW CREATE VIEW does not support showing CREATE TABLE statement with identifier %s.",
```

Comment:
I think we could make this more helpful for the user:

```suggestion
                            "SHOW CREATE VIEW is only supported for views, but '%s' is a table. Please use SHOW CREATE TABLE instead.",
```